### PR TITLE
Fixes #25480 - corrected query to get active pool count

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

File: app/models/katello/concerns/organization_extensions.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@ def redhat_provider
         end
 
         def active_pools_count
-          self.pools.count { |p| p[:activeSubscription] == true }
+          Katello::Pool.get_for_owner(self).count { |p| p['activeSubscription'] == true }
```
[the remark this answers is not in the document]
Comment (kgaikwad) [reply app/models/katello/concerns/organization_extensions.rb]: @jlsherrill,
Do you mean to store this value to pool table itself while adding pool to our DB?

[the remark this answers is not in the document]
Comment (jlsherrill) [reply app/models/katello/concerns/organization_extensions.rb]: yep, exactly

**Comment by beav** [comment]
adding WoC per https://github.com/Katello/katello/pull/7839#discussion_r234304768

File: app/models/katello/concerns/organization_extensions.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kgaikwad) [reply app/models/katello/concerns/organization_extensions.rb]: @jlsherrill,
Thank you for giving pointers. It solved all my queries.
I found one more [place](https://github.com/Katello/katello/blob/master/app/models/katello/glue/candlepin/pool.rb#L19) where it uses 'activeSubscriptions' to get active status.

Please let me know where it also needs modification.

[the remark this answers is not in the document]
Comment (kgaikwad) [reply app/models/katello/concerns/organization_extensions.rb]: @jlsherrill, @beav,

Updated pull-request with change `organization.pools.count`. In addition to this skipped "unmapped_guests" records from `organization.pools`.

**Comment by beav** [comment]
Is unmapped guest for hosts that get temporary subs until virt-who checkin occurs?

**Comment by johnsonm325** [comment]
@kgaikwad I was doing some work on this and just as I was about to push up my branch, I saw that you had already fixed it. :smile: 

I've tested out your fix and see that it's working similar to mine, but yours is not having to communicate with candlepin. That also makes it easier to test. Do you mind implementing a test for this as well? Lucky for you, I've already started something: katello/test/models/concerns/organization_extensions_test.rb

```
# encoding: utf-8

require 'katello_test_helper'

module Katello
  class OrganizationExtensionsTest < ActiveSupport::TestCase
    def setup
      User.current = User.find(users(:admin).id)
      set_default_location
      @katello_id = "KT_Org_Env_View_1"

      @org = get_organization
      @org.label = @org.label.tr(' ', ' ')
    end

    def test_active_pools_count
      pool = @org.active_pools_count
      assert_equal pool, 3

      @org.pools.first.update_attribute(:unmapped_guest, true)
      pool = @org.active_pools_count
      assert_equal pool, 2
    end
  end
end

```

**Comment by kgaikwad** [comment]
> @kgaikwad I was doing some work on this and just as I was about to push up my branch, I saw that you had already fixed it. 
> 
> I've tested out your fix and see that it's working similar to mine, but yours is not having to communicate with candlepin. That also makes it easier to test. Do you mind implementing a test for this as well? Lucky for you, I've already started something: katello/test/models/concerns/organization_extensions_test.rb
> 
> ```
> # encoding: utf-8
> 
> require 'katello_test_helper'
> 
> module Katello
>   class OrganizationExtensionsTest < ActiveSupport::TestCase
>     def setup
>       User.current = User.find(users(:admin).id)
>       set_default_location
>       @katello_id = "KT_Org_Env_View_1"
> 
>       @org = get_organization
>       @org.label = @org.label.tr(' ', ' ')
>     end
> 
>     def test_active_pools_count
>       pool = @org.active_pools_count
>       assert_equal pool, 3
> 
>       @org.pools.first.update_attribute(:unmapped_guest, true)
>       pool = @org.active_pools_count
>       assert_equal pool, 2
>     end
>   end
> end
> ```

Thank you @johnsonm325 for testing PR changes as well as for sharing test. Updated the pull-request.

**Comment by kgaikwad** [comment]
> Is `unmapped_guest` just to not count hosts that get temporary subs until virt-who checkin occurs?

yes but I am not sure about whether need to consider those subs as active?
Even with previous code, it was fetching pools count from candlepin using two conditions i.e `add_future = true and unmapped_guests_only:!true` and that's why I added condition on `organization.pools`.

**Comment by beav** [comment]
> > Is `unmapped_guest` just to not count hosts that get temporary subs until virt-who checkin occurs?
> 
> yes but I am not sure about whether need to consider those subs as active?
> Even with previous code, it was fetching pools count from candlepin using two conditions i.e `add_future = true and unmapped_guests_only:!true` and that's why I added condition on `organization.pools`.
> @beav, please correct me if anything missing from my side.

It is ok to not count those as active IMO. Mostly I was asking just to confirm that I understood the query :cat:

**Comment by beav** [comment]
code looks good, unit test looks good, and patch was tested by @johnsonm325.

ack thanks @kgaikwad !

**Comment by johnsonm325** [comment]
@kgaikwad Tested this myself again and the numbers match up when you check for active status on the pool. Everything looks good to me. ACK!

**Comment by jturel** [comment]
@jlsherrill you had some feedback on this PR - look good to you?

**Pull request merged**